library/alloc/src/raw_vec.rs

```diff
@@ -114,7 +114,10 @@ impl<T> RawVec<T, Global> {
     #[must_use]
     #[inline]
     pub fn with_capacity(capacity: usize) -> Self {
-        handle_reserve(Self::try_allocate_in(capacity, AllocInit::Uninitialized, Global))
+        match Self::try_allocate_in(capacity, AllocInit::Uninitialized, Global) {
+            Ok(res) => res,
+            Err(err) => handle_reserve_error(err),
+        }
```

Comment:
I feel like this would be more concise and more readable with `unwrap_or_else`.

Reply:
In this file there's a comment:

            // We avoid `unwrap_or_else` here because it bloats the amount of
            // LLVM IR generated.

Reply:
Ah, that wasn't visible in the diff, thanks.